eth/downloader/queue.go

```diff
@@ -349,66 +341,78 @@ func (q *queue) Schedule(headers []*types.Header, from uint64) []*types.Header {
 
 // Results retrieves and permanently removes a batch of fetch results from
 // the cache. the result slice will be empty if the queue has been closed.
+// This is 'thread-safe', but assumes that there are not two simultaneous
+// callers to Results (both will modify q.resultSize)
```

Comment:
This comment should be: "Results can be called concurrently with Deliver and ...."